libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/NavigationEventDispatcherTest.java

```diff
@@ -309,24 +317,49 @@ public void setNavigationMetricListener_didGetSet() throws Exception {
 
   @Test
   public void setNavigationMetricListener_didNotGetTriggeredUntilArrival() throws Exception {
-    RouteLeg lastLeg = route.legs().get(route.legs().size() - 1);
-    int lastStepIndex = lastLeg.steps().indexOf(lastLeg.steps().get(lastLeg.steps().size() - 1));
-
+    DirectionsRoute route = buildTestDirectionsRoute();
+    int first = 0;
+    String instruction = "";
+    RouteLeg routeLeg = route.legs().get(first);
+    List<LegStep> routeSteps = routeLeg.steps();
+    int currentStepIndex = routeSteps.size() - 2;
+    int upcomingStepIndex = routeSteps.size() - 1;
+    LegStep currentStep = routeSteps.get(currentStepIndex);
+    LegStep upcomingStep = routeSteps.get(upcomingStepIndex);
+    RouteProgress routeProgress = buildRouteProgress(first, route, currentStep, upcomingStep);
+    BannerInstructionMilestone bannerInstructionMilestone = mock(BannerInstructionMilestone.class);
+    List<BannerInstructions> currentStepBannerInstructions = currentStep.bannerInstructions();
+    buildBannerInstruction(first, bannerInstructionMilestone, currentStepBannerInstructions);
     navigationEventDispatcher.addMetricEventListeners(eventListeners);
     navigationEventDispatcher.addMetricArrivalListener(arrivalListener);
 
-    RouteProgress routeProgressDidNotArrive = buildTestRouteProgress(route, 100,
-      100, 100, lastStepIndex, 0);
+    navigationEventDispatcher.onMilestoneEvent(routeProgress, instruction, bannerInstructionMilestone);
+
+    verify(arrivalListener, times(0)).onArrival(routeProgress);
```

Comment:
Shouldn't this be broken into two different tests? Since you're verifying two different things